Development/OIDC.md

```diff
@@ -1,169 +1,34 @@
-* **successRedirect** is the URL passed to the OIDC provider instructing it to redirect the user on success. Here we set it to the a development environment's user page.
-* **failureRedirect** is the URL passed to the OIDC provider instructing it to redirect the user on failed authentication. Here we set it to the a development environment's user page login page.
-* **failureFlash** is currently unsupported.
-* **session** is not needed, since the frontend authenticates outside of sessions.
-* **authorizationURL** is specific to your OIDC provider
-* **tokenURL** is specific to your OIDC provider
-* **userInfoURL** is specific to your OIDC provider
-* **clientID** is specific to your OIDC provider
-* **clientSecret** is specific to your OIDC provider
-* **scope** is specific to your OIDC provider
-* **loginCallback** optional name of a [callback](#configure-callback) function to run to add profile information to the UserIdentity
-
-#### Session Secret 
-The nodejs module `passport-openidconnect` requires the use of the `express-session` module
-to create a login session for the user.
- This requires a _session secret_ to be configured in `config.local.js`:
- ```
- module.exports = {
-  expressSessionSecret: "thisIsSuperSecretForUserSession"
- ```
-
-## SciCat Frontend Configuration 
-
-The default login page in the frontend provides a username/password form that is used both for local and LDAP/AD authentication. When using and OIDC authentication provider, SciCat will not ask the user for credential directly. Instead, the user will be redirected to the OIDC provider to authenticate. So, the frontend provides two configuration settings to modify the login page appropriately.
-
-These setting are accomplished by modifying the frontend [environment document](./Environment.md). 
-
-```javascript
-  
-  loginFormEnabled: false,
-  oAuth2Endpoints: [
-    {displayText: "Google", displayImage: "../../../assets/images/btn_google_light_normal_ios.svg", authURL: "auth/google"}]
-
-```
-
-
-* **loginFormEnabled** sets whether to display the username/password form in the login page
-* **oAuth2Endpoints** provides information that the frontend uses to display "Sign in with ..." buttons. Note that this is an array, multiple buttons for multiple OIDC providers can be configured.
-  * **displayText** sets the name of the provider to display in the button. In this case, the button will show "Sign In With Google"
-  * **displayImage** defines an image to display in the button. The image will end up being 24px tall. It is recommended that the image be in SVG format.
-  * **authURL** defines the relative path that the user will be redirected to when they click the button. Note that this maps to the `authPath` setting described above.
-
-## Backend Code Hooks
```

Comment:
It's on my list to rewrite this bit, since we're the facility who needs it the most!

Reply:
apologies - I have put this back for your editing